zrml/court/src/lib.rs

```diff
@@ -190,8 +208,130 @@ mod pallet {
         }
 
         // Retrieves a juror from the storage
-        fn juror(account_id: &T::AccountId) -> Result<Juror<BalanceOf<T>>, DispatchError> {
-            Jurors::<T>::get(account_id).ok_or_else(|| Error::<T>::JurorDoesNotExists.into())
+        fn mutate_juror<F>(account_id: &T::AccountId, mut cb: F) -> DispatchResult
+        where
+            F: FnMut(&mut Juror<BalanceOf<T>>) -> DispatchResult,
+        {
+            Jurors::<T>::try_mutate(account_id, |opt| {
+                if let Some(el) = opt {
+                    cb(el)?;
+                } else {
+                    return Err(Error::<T>::JurorDoesNotExists.into());
+                }
+                Ok(())
+            })
+        }
+
+        // Slashes 20% of staked funds and removes them from the list of jurors
+        fn punish_tardy_jurors(
+            requested_jurors: &mut Vec<(T::AccountId, T::BlockNumber)>,
+        ) -> DispatchResult {
+            let mut i = 0;
+            while i < requested_jurors.len() {
+                let should_remove = {
+                    // `get` will never panic
+                    let (ai, _) = requested_jurors.get(i).unwrap();
+                    let juror = Self::juror(ai)?;
+                    matches!(juror.status, JurorStatus::Tardy)
+                };
+                if should_remove {
+                    // `remove` will never panic
+                    let (ai, _) = requested_jurors.remove(i);
+                    let reserved = CurrencyOf::<T>::reserved_balance(&ai);
+                    // Division will never overflow
+                    let slash = reserved / BalanceOf::<T>::from(5u8);
```

Comment:
We should move the hardcoded slash percentage of 1/5 to a config value, e.g. `TARDY_PUNISHMENT` .

Reply:
Done